### ISSUE TITLE
Perma-red: browser_aboutHome.js The expected URL was loaded - Got about:home

### ISSUE THREAD

**Issue by piatra** [mozilla/activity-stream#3255]
```
waitForDocLoadAndStopIt: The expected URL was loaded - Got about:home, expected http://mochi.test:8888/
```

in `browser/base/content/test/about/browser_aboutHome.js`

**Comment by Mardak** [comment]
Oddly clicking search when input is empty seems to reload the page on first click. Second click on the button loads the search page. ?

**Comment by Mardak** [comment]
Oh. This looks to be a `browser.newtab.preload` issue. The second loaded page is not preloaded, and things work.

**Comment by piatra** [comment]
I still see the test failing with preload set to off.

**Comment by Mardak** [comment]
Do you see this from manual testing or only via the test? I can't seem to reproduce on my usual profile with the official nightly builds when setting aboutHome.enabled pref

**Comment by piatra** [comment]
This is via the test.

**Comment by sarracini** [comment]
can you post a link to the try run that has this test failure? i can't reproduce this failure locally - though for me the test just hangs and times out so that's not great either

**Comment by Mardak** [comment]
My running of the test also seems to just time out:
```
./mach mochitest --setpref=browser.newtabpage.activity-stream.aboutHome.enabled=true browser/base/content/test/about/browser_aboutHome.js
2 INFO TEST-START | browser/base/content/test/about/browser_aboutHome.js
4 INFO TEST-PASS | browser/base/content/test/about/browser_aboutHome.js | Should disable the onboarding tours - 
13 INFO TEST-UNEXPECTED-FAIL | browser/base/content/test/about/browser_aboutHome.js | Test timed out - 
14 INFO TEST-OK | browser/base/content/test/about/browser_aboutHome.js | took 208108ms
15 INFO TEST-UNEXPECTED-FAIL | browser/base/content/test/about/browser_aboutHome.js | Found a tab after previous test timed out: about:home - 
```

**Comment by piatra** [comment]
@Mardak @sarracini it times out because of the snippet tests. The file is a mix of snippets, search and telemetry tests. By default it times out on waiting for about:home specific events that never happen. Try commenting out and leaving only that test.

**Comment by Mardak** [comment]
This specific line's failure seems to be fixed with #3260.

**Comment by piatra** [comment]
Fixed for me with #3260
